extensions/wikia/AdEngine/js/template/bfaaMobile.js

```diff
@@ -103,43 +59,24 @@ define('ext.wikia.adEngine.template.bfaaMobile', [
 			});
 		}
 
-		if (params.videoTriggerElement && params.videoAspectRatio) {
-			videoAdFactory.init().then(function () {
-				try {
-					var video = videoAdFactory.create(
-						slotSizes.width,
-						slotSizes.videoHeight,
-						adSlot,
-						{
-							src: 'gpt',
-							pos: params.slotName,
-							uap: params.uap,
-							passback: 'vuap'
-						}
-					);
-
-					page.classList.add('vuap-loaded');
+		if (uapVideo.isVideoAd(params)) {
+			uapVideo.init()
+				.then(function () {
+					var video = uapVideo.loadVideoAd(params, adSlot, imageContainer);
 
-					window.addEventListener('resize', adHelper.throttle(function () {
-						slotSizes = getSlotSize(params);
-						video.resize(slotSizes.width, slotSizes.videoHeight);
-					}));
-
-					params.videoTriggerElement.addEventListener('click', function () {
-						video.play(showVideo, hideVideo);
+					video.addEventListener(win.google.ima.AdEvent.Type.LOADED, function () {
```

Comment:
:+1: :-1: 

:+1: - for removing `onResize.bind(null, params.aspectRatio)`
:-1: - for exposing the implementation details (a fact that we use ima)

Did you think about different approaches (at least to have our own “enum”)?